dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/data/file/FileWriter.kt

```diff
@@ -0,0 +1,113 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-2019 Datadog, Inc.
+ */
+
+package com.datadog.android.core.internal.data.file
+
+import android.annotation.TargetApi
+import android.os.Build
+import android.util.Base64 as AndroidBase64
+import com.datadog.android.core.internal.data.Orchestrator
+import com.datadog.android.core.internal.data.Writer
+import com.datadog.android.core.internal.domain.Serializer
+import com.datadog.android.core.internal.threading.LazyHandlerThread
+import com.datadog.android.log.internal.utils.sdkLogger
+import java.io.File
+import java.io.FileNotFoundException
+import java.io.IOException
+import java.util.Base64 as JavaBase64
+
+internal class FileWriter<T : Any>(
+    private val fileOrchestrator: Orchestrator,
+    rootDirectory: File,
+    private val serializer: Serializer<T>
+) : LazyHandlerThread(THREAD_NAME),
+    Writer<T> {
+
+    // TODO: RUMM-148 Clean this code from here and move it into the FileOrchestrator eventually
+    private val writeable: Boolean = if (!rootDirectory.exists()) {
+        rootDirectory.mkdirs()
+    } else {
+        rootDirectory.isDirectory
+    }
+
+    init {
+        if (!writeable) {
+            sdkLogger.e(
+                "$TAG: Can't write logs on disk: directory ${rootDirectory.path} is invalid."
```

Comment:
Change the error message to "Can't write **data** on disk …". Same goes with all error messages in this file (and most probably in the `FileReader` class too).